src/Http/Routing/src/SuppressApiMetadata.cs

```diff
@@ -0,0 +1,23 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using Microsoft.AspNetCore.Http;
+
+namespace Microsoft.AspNetCore.Routing
+{
+    /// <summary>
+    /// Specifies an endpoint name in <see cref="Endpoint.Metadata"/>.
+    /// </summary>
+    /// <remarks>
+    /// Endpoint names must be unique within an application, and can be used to unambiguously
+    /// identify a desired endpoint for URI generation using <see cref="LinkGenerator"/>.
+    /// </remarks>
+    public sealed class SuppressApiMetadata : ISuppressApiMetadata
```

Comment:
Just curious, why _suppress_ rather than _hide_ or _ignore_? As this isn't specific to the API Explorer stuff as it's just routing metadata, it could maybe suggest it suppresses the _entire_ API (i.e. make it non-usable), rather than just its presence in API documentation?

Reply:
Hmmmm. Yeah. My mind gravitated towards suppress because this endpoint prevents/suppresses the API explorer pipelines from emitting metadata for this endpoint.

TBH, `IgnoreApi`, `HideApi`, and `SuppressApi` all struck me as having some degree of vagueness. Perhaps the right thing to do is name this `HideApiInApiExplorer` or something to that affect.

I'll bring this point up during API review and see what thoughts folks have.